File: EmsBMSReports/DlgNewGroup.cs
using System;
using System.Windows.Forms;

namespace EmsBMSReports
{

    public partial class DlgNewGroup
    {
        public DlgNewGroup()
        {
            InitializeComponent();
        }

        private void OK_Button_Click(object sender, EventArgs e)
        {
            int localGetGroupID() { string argsGroupName = tGroupName.Text; var ret = MactusReportLib.MactusReportLib.GetGroupID(ref argsGroupName); tGroupName.Text = argsGroupName; return ret; }

            if (localGetGroupID() > 0)
            {
                MessageBox.Show("Group Name Already Exist.", "Add NEw Data Trend Group", MessageBoxButtons.OK, MessageBoxIcon.Error);
            }
            else
            {
                string argsGroupName = tGroupName.Text;
                MactusReportLib.MactusReportLib.AddNewDataTrendGroup(ref argsGroupName);
                tGroupName.Text = argsGroupName;
                DialogResult = DialogResult.OK;
                Close();
            }
        }

        private void Cancel_Button_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.Cancel;
            Close();
        }

        private void tGroupName_TextChanged(object sender, EventArgs e)
        {
            if (tGroupName.Text.Length <= 2)
            {
                OK_Button.Enabled = false;
            }
            else
            {
                OK_Button.Enabled = true;
            }
        }
    }
}

File: ReportsConfig/DlgNewGroup.vb
Imports System.Windows.Forms
Imports MactusReportLib

Public Class DlgNewGroup

    Private Sub OK_Button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles OK_Button.Click
        If GetGroupID(tGroupName.Text) > 0 Then
            MessageBox.Show("Group Name Already Exist.", "Add NEw Data Trend Group", MessageBoxButtons.OK, MessageBoxIcon.Error)
        Else
            AddNewDataTrendGroup(tGroupName.Text)
            Me.DialogResult = System.Windows.Forms.DialogResult.OK
            Me.Close()
        End If
    End Sub

    Private Sub Cancel_Button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Cancel_Button.Click
        Me.DialogResult = System.Windows.Forms.DialogResult.Cancel
        Me.Close()
    End Sub

    Private Sub tGroupName_TextChanged(sender As Object, e As EventArgs) Handles tGroupName.TextChanged
        If tGroupName.Text.Length <= 2 Then
            OK_Button.Enabled = False
        Else
            OK_Button.Enabled = True
        End If
    End Sub
End Class
